app/assets/javascripts/lib/views/d3_chart_view.coffee

```diff
@@ -26,8 +26,15 @@ class @D3ChartView extends BaseChartView
       @check_merit_enabled()
       @drawn = App.settings.merit_order_enabled()
 
+    @display_empty_message()
     @refresh()
 
+  is_empty: =>
+    false
+
+  display_empty_message: =>
+    EmptyChartMessage.display.call(this)
```

Comment:
Instead of `call(this)`, which I think is really ambiguous in `EmptyChartMessage`, what do you think of instead doing `EmptyChartMessage.display(this)`, and have `display` take an explicit parameter instead?

```js
var EmptyChartMessage = (function () {
        display: function (chart) {
            var container = $(chart.container_selector()),
                wrapper   = template.call(chart, container),
                isEmpty   = chart.is_empty();

                // ...
        }
}
```